doc/ref_cert/lfn/chapters/chapter02.md

```diff
@@ -14,38 +14,58 @@
+* Network intensive (N) profile offerring low latency and high networking throughout
+Similarly, different NFVI vendors may specialise in different hardware profiles and some may specialise in both VNFs and NFVI. To cater to different needs from multiple NFVI vendors, CNTT allows different types of NFVI certification based on their types of [profile](../../../ref_model/chapters/chapter02.md#2.3)
+ * Certify Vendor NFVI Hardware solution: This allows for certification of only NFVI.
+ * Certify Vendor NFVI Hardware and Software Solution: This allows for certification for NFVI running a particular VNF.
 
 <a name="2.5"></a>
 ## 2.5 Compliance, Verification, and Certification
-
-Define the framework and process for NFVI certification..
+The below set of steps define the compliance, verification and certification process for NFVI
+* Based on VNF's requirements, NFVI profile is selected - B, C, N
+* NFVI readiness is checked for certification.
+* The test VNFs are on-boarded using automation scripts on the NFVI.
+* VNF on-boarding is validated by running functional tests to ensure that the on-boarding is successful.
+* VNF performance tests are executed and NFVI KPIs are recorded during the tests.
+* KPI comparison is run to compare NFVI KPIs with Golden KPIs, which serve as a reference for NFVI certification.
+* If NFVI KPIs meet Golden KPIs, NFVI is certified and granted a certification badge.
+* If NFVI KPIs do not meet Golden KPIs, no certification is provided.
 
 <a name="2.6"></a>
 ## 2.6 Entry & Exit Criteria
+**Entry criteria**: Before entering into NFVI certification, NFVI needs to satisfy the following requirements as entry pass:
+* Lab Requirements*: The NFVI lab needs to be setup according to RA1/RA2 as defined by CNTT and should be in the required state.
 
-Provide a list of entry and exit criteria to be used for NFVI certifications..
+**Exit criteria**: NFVI certification testing should completed with following exit criteria:
+* All mandatory test cases should pass.
+* Test results collated, centralized, and normalized, with a final report generated showing status of the test scenario/case (e.g. Pass, Fail, Skip, Measurement Success/Fail, etc), along with traceability to a functional, or non-functional, requirement.
 
 <a name="2.7"></a>
 ## 2.7 Frameworks
+The NFVI certification framework deals with the process of testing NFVI in below three areas:
+* Compliance: The NFVI needs to comply to CNTT RA1/RA2.
+* Validation: Validation deals with the ability of NFVI to respond to Cloud APIs and interfaces.
+* Performance: Performance deals with running tests on NFVI depending on the NFVI profile and collecting KPIs.
 
-List out the various frameworks requirements needed for certifications
+The NFVI KPIs are compared with Golden KPIs, which serve as a reference for NFVI certification. If NFVI KPIs meet Golden KPIs, NFVI is certified and granted a certification badge. If NFVI KPIs do not meet Golden KPIs, no certification badge is provided.
```

Comment:
Add the following to this section:

1. A comment about why automated test validation is critical, needed, and will be a governing principle.  Reference examples of Functest, if needed.

2. Reference a list of Test Categories that we will define as part of the framework, which will help collate test results and identify the test type.  Add these to this section:

1. On-Boarding (ONAP MANO, NFVi Agnostic)
2. Instantiation, Deletion, Recent Change (ONAP MANO, NFVi Agnostic)
3. VNF Functional Testing
4. Security Testing
5. Charging / Revenue Assurance Verification
6. MicroServices Support
7. Closed Loop Testing
8. VNF Coexistence (ETSI NFV-TST001 “Noisy Neighbor”)
9. VNF Interactions with Extended NFVi Topology
10. VNF Interactions with Complex NFVi (Akraino)
11. Scalability Testing
12. HA Testing
13. Fault Recovery Testing
14. PM/KPI/Service Assurance Testing
 
3. Also, include a section, or mention that our Test Harnesses will adhere to these standards and strategies:

1. Standardized test methodology / flow
2. Working test bed reference design
3. Standardized Cloud-based facilities (storage, IAM, etc.)
4. Standard ONAP Network and Service Models, with identified VNF-specific parameters
5. Robot library to enable Data-Driven testing of On-Boarding, Instantiation, etc.
6. Sample VNF, CSAR and Robot Test Cases
7. Standardized base Test Plan and Test Case suite
8. Standardized certification criteria
9. Integration with Dovetail and OVP flow (code, docs, cert criteria, etc.)
10. Real, usable documentation
11. “Just add Water” deployment vehicle

Also, mention that our goal is to have defined Test Categories | Cases... add this info..

**Test Categories | Cases**

The following five test categories have been identified as minimal testing required to verify NFVI interoperability to satisfy the needs of VNF developer teams.

**Test Categories**

1. Baremetal - validations
- We will validate Control and compute nodes hardware.
2. VNF Interoperability - validations
- After VNF on boarded we are validating end to end openstack resources like Tenant, Network (L2/L3), CPU Pining, security policies, Affinity anti-affinity roles and flavours etc.
3. Compute Component - validations
- Validate/Document VMs status and connectivity result after performing each of listed steps. Best candidate for this testing would be identify compute node that holds VMs which has l2 and l3 connectivity.
4. Control Plane Component- validations
- We are validating RabbitMQ, Ceph, Mariadb and Openstack components like nova, glance, heat, keystone API and resillency test.
5. Security - see Chapter 7 for complete list
- Validating user RBAC Roles and User group policies.

Respective test cases can be found in the [where of] Chapter [what].

Also, add **Test Results**

Test suites will be categorized as functional or performance based. Results reporting will be communicated as a boolean (pass/fail). The pass/fail determination for performance-based test cases will be made by comparing results against a baseline. Example performance-based metrics include, but are not limited to: resource utilization, response times, latency, and sustained throughput per second (TPS).

Include this comment about **Measurements**,

Measurements and respective certification(s). e.g. OVP Testing Ecosystem badge (with link to certification with posted results)

Test validations will be corroborated, and confirmed, with direct comparison between measured results and documented non-functional requirements (NFRs) for applications, hardware and software configuration settings, and host systems. Throughput, latency, concurrent connections/threads, are all examples of non-functional requirements which specify criteria which can be used to judge the operation of a system, rather than specific behaviours of the application which are defined by functional requirements.

This section attempts to summarize a categorical list of metrics used for test validations. For a complete list of metrics, and requirements, please visit Chapter 4 of the Reference Model.

Reply:
Updated. I have added test categories in Section 2.7.1.1 Test Categories.
Regarding 3. On-boarding ... 28 .“Just add Water” deployment vehicle -> I am not sure if these needs to be a part of this chapter. 
What is your idea ?

Reply:
@msidana - Manik, I'll work on this in a little while.  I believe I can edit, or add clarity.  Will update when edits are available for review.  In the meantime, will keep this conversation thread open.  Thanks, Mike

Reply:
@msidana - Manik, I re-wrote section 2.7.  Please review the logic and content.  Please edit, or add missing content.  If satisfied, please close/resolve this thread.  Thanks, Mike

Reply:
Very well drafted @mf4716 . Couldn't be any better. Resolving. Thanks.